shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/impl/EncryptProjectionTokenGenerator.java

```diff
@@ -63,10 +75,19 @@ protected boolean isGenerateSQLTokenForEncrypt(final SQLStatementContext sqlStat
     @Override
     public Collection<SubstitutableColumnNameToken> generateSQLTokens(final SelectStatementContext selectStatementContext) {
         ProjectionsSegment projectionsSegment = selectStatementContext.getSqlStatement().getProjections();
-        // TODO process multiple tables
-        String tableName = selectStatementContext.getAllTables().iterator().next().getTableName().getIdentifier().getValue();
-        return getEncryptRule().findEncryptTable(tableName).map(
-            encryptTable -> generateSQLTokens(projectionsSegment, tableName, selectStatementContext, encryptTable)).orElseGet(Collections::emptyList);
+        Collection<SubstitutableColumnNameToken> result = new HashSet<>();
+        Collection<String> tableNames = selectStatementContext.getTablesContext().getTableNames();
+        for (String each : tableNames) {
+            Optional<EncryptTable> encryptTable = getEncryptRule().findEncryptTable(each);
+            if (encryptTable.isPresent()) {
+                Collection<SubstitutableColumnNameToken> sqlTokens = generateSQLTokens(projectionsSegment, each, selectStatementContext, encryptTable.get());
+                result.addAll(sqlTokens);
+            }
+        }
+        if (selectStatementContext.isContainsJoinQuery()) {
```

Comment:
@cheese8 Why do you need to process the associated query separately instead of in the general generateSQLTokens method above?

Reply:
changed and fixed